test/math.jl

```diff
@@ -192,6 +192,25 @@ end
 @test exp2(Float16(2.)) ≈ exp2(2.)
 @test log(e) == 1
 
+@testset "check exp2(::Int) matches exp2(::Float)" begin
```

Comment:
see https://github.com/JuliaLang/julia/pull/17165 for what's needed to use testsets in base. they're too noisy at the moment, can't use them just yet